src/main/drivers/pwm_output_dshot_hal.c

```diff
@@ -82,6 +101,28 @@ FAST_CODE void pwmCompleteDshotMotorUpdate(uint8_t motorCount)
 {
     UNUSED(motorCount);
 
+    /* If there is a dshot command loaded up, time it correctly with motor update*/
+    if (dshotCommandControl.nextCommandAt) {
```

Comment:
This code really shouldn't have gone into dshot update function which is placed into `ITCM-RAM` :-(

Reply:
The problem is that this has to be in the Dshot output loop. Performance isn't a factor when sending Dshot commands, so that's another reason to encapsulate the actual handling of the command buffer in a (not inlined) function that isn't put into `ITCM-RAM`. Can you advise on how to best do this @DieHertz?

Reply:
The checking for dshot commands could happen at writeMotors() in mixer.c.  It just needs control over when pwmCompleteMotorUpdate() happens to get the timing right.  It didn't seem like the right spot at the when I started looking at this (the dshot command check should happen in dshot functions), but it would keep that extra code out of the dshot update function.

Reply:
Any thoughts @DieHertz @mikeller ?

Reply:
I'd go for leaving the `if (dshotCommandControl.nextCommandAt) {` in the `FAST_CODE` functions, but extracting the rest of the command handling functions into `NOINLINE` functions, and calling it there. This way, the size increase for ITCM-RAM is minimised.